packages/core/src/UserState.ts

```diff
@@ -90,10 +95,13 @@ export default class UserState {
         }
         this._id = id
         this._prover = prover
+        this._chainId = -1 // need to be setup in async function
     }
 
     async start() {
         await this.sync.start()
+        const { chainId } = await this.sync.provider.getNetwork()
```

Comment:
Maybe use `this._checkChainId`?

Reply:
Thanks 🙏🏻
[3d0902b](https://github.com/Unirep/Unirep/pull/547/commits/3d0902b9534b5a456177cf90c394117957f9a23d)